### ISSUE TITLE
Downloading data with null value as JSON causes an Internal Server Error

### ISSUE THREAD

**Issue by bpeXX** [ckan/ckan#7037]
**CKAN version**
2.8.4 and 2.8.10

**Describe the bug**
When downloading data (which is stored in the datastore) as JSON, an internal server error occurs if the data contains a null value. The error occurs when each row has at least one column with a null value.

**Steps to reproduce**
The bug can be reproduced via ckan API:
1. Create resource: http://myserver:myport/api/3/action/resource_create
    with body form data: package_id (ID of an existing dataset) and name (name of resource)
2. Create datastore and add data: http://myserver:myport/api/3/action/datastore_create
    with body raw data:
`{
  "resource_id": "54317d3a-3a26-4911-911d-57c615639756",
  "force": "true",
  "fields": [
    {
      "id": "id",
      "type": "numeric"
    },
    {
      "id": "name",
      "type": "text"
    }
  ],
  "primary_key": "id",
  "records": [
    {
      "id": "1",
      "name": null
    }
  ]
}`
3. Download data as JSON: http://myserver:myport/datastore/dump/54317d3a-3a26-4911-911d-57c615639756?format=json

**Expected behavior**
A download should be possible without error. In this particular case the resulting json file will be empty. (In case of multiple rows of data, ckan already ignors the rows with null values and downloads the rest of data successfully.)

**Additional details**
Stack trace:
![image](https://user-images.githubusercontent.com/111954802/186346787-383ff280-ad23-416e-a195-a8501b54da4a.png)

**Comment by robin-NEC** [comment]
Hi @wardi @bpeXX , should we allow users to upload files with `null` values? I tried to reproduce the issue but received a different error that is associated with the #6713  issue, I've fixed issue #6713 and will raise the PR soon.

 But I found that when downloading a resource in json format, [this](https://github.com/ckan/ckan/blob/master/ckanext/datastore/backend/postgres.py#L1372-#L1378) query runs. I found that when concatenating with the `||` operator, the values from a row that contains a `null` value, the output for that particular row changes into a blank row, I've tested with multiple queries containing  `null` values, I believe this is the default behavior of PostgreSQL.

We can have a different approach where we can replace the `null` value with another value, but that would be an alteration of the file itself.
I believe a better approach would be to not allow the users to upload files containing `null` values. 

@wardi, I'd like to know your opinion on this.
